streamlit/streamlit
# Cleanup pre-commit setup

## Describe your changes

This PR applies a couple of improvements & cleanups related to pre-commit:

- Remove pre commit install from shared `make_init` action.
- Move enforcement of pre-commit rules from `js-tests` to dedicated workflow (takes < 1 min) -> missing license headers in python files will not cause the js-tests workflow to fail anymore.
- Use make commands in js-tests workflow
- Remove eslint, typecheck-app, typecheck-lib from pre-commit since these checks are not actually run as part of the normal pre-commit.
- Update versions in pre-commit config.
- Add some additional pre-commit checks: check-json, check-yaml, check-toml, check-added-large-files.
- Fix some json / yml issues.

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.
